sail/src/main/java/org/apache/rya/rdftriplestore/inference/InferenceEngine.java

```diff
@@ -416,22 +419,57 @@ private void refreshHasValueRestrictions(Map<Resource, URI> restrictions) throws
         }
     }
 
-    private static Vertex getVertex(Graph graph, Object id) {
-        Iterator<Vertex> it = graph.vertices(id.toString());
+    private void refreshHasSelfRestrictions(final Map<Resource, URI> restrictions) throws QueryEvaluationException {
+        hasSelfByType = new HashMap<>();
+        hasSelfByProperty = new HashMap<>();
+
+        restrictions.forEach((type, property) -> {
+            try {
+                final CloseableIteration<Statement, QueryEvaluationException> iter = RyaDAOHelper.query(ryaDAO, type, HASSELF, null, conf);
+                try {
+                    if (iter.hasNext()) {
+                        Set<URI> typeSet = hasSelfByType.get(type);
+                        Set<Resource> propSet = hasSelfByProperty.get(property);
+
+                        if (typeSet == null) {
+                            typeSet = new HashSet<>();
+                        }
+                        if (propSet == null) {
+                            propSet = new HashSet<>();
+                        }
+                        typeSet.add(property);
+                        propSet.add(type);
+
+                        hasSelfByType.put(type, typeSet);
+                        hasSelfByProperty.put(property, propSet);
+                    }
+                } catch (final QueryEvaluationException e) {
+                    if (iter != null) {
+                        iter.close();
+                    }
+                }
+            } catch (final QueryEvaluationException e) {
```

Comment:
We shouldn't silently swallow this Exception.  Maybe don't use a Java 8 lambda here since they're poor at handling throwing checked Exceptions.  There are some workarounds for throwing an Exception but most of them aren't ideal.